subgraphs/_reference_/src/sdk/protocols/bridge/pool.ts

```diff
@@ -252,6 +261,80 @@ export class Pool {
     this.save();
   }
 
+  /**
+   * Updates the net value exported for this pool to the given value.
+   * Will also update the protocol's net value exported based on the change in this pool's.
+   *
+   * @param newNetValueExported The new net value exported for this pool.
+   */
+  setNetValueExportedUSD(newNetValueExported: BigDecimal): void {
+    const delta = newNetValueExported.minus(this.pool.netValueExportedUSD);
+    this.addNetValueExportedUSD(delta);
+  }
+
+  /**
+   * Adds the given delta to the net value exported for this pool.
+   * Will also update the protocol's net value exported based on the change in this pool's.
+   * Since at the protocol level valueExported is broken down into totalValueExported and totalValueImported,
+   * it is a bit tricky to update those. In order to do so, we deduct the current netValueExported from
+   * the protocols, recalculate the pool's netValueExported, and then add it back again to the protocol.
+   *
+   * @param delta The change in net value exported for this pool.
+   */
+  addNetValueExportedUSD(delta: BigDecimal): void {
```

Comment:
Will the logic be more clear if we track totalValueImported/totalValueExported - adding/subtracting when transferIn/Out - and derive netValueExported from them?

Reply:
I think we can’t do that. Because value imported/exported is not exactly volume. It’s the value of all assets currently living in the bridge  (or away) for a given pool.

For lock/release pools, it will be the value of all assets locked. And this value can change both on transferin/out but also if the price of the asset changes.

Same for mint/burn pools, it will be the value of all supply minted.

For lock/release, totalValueImported will always be 0. And totalValueExported will be the value of locked assets (goes up and down with transfers and with price changes).
For mint/burn, totalValueExported will always be 0. And totalValueImported will be the value of supplies minted (goes up and down with transfers and price changes).

Reply:
makes sense.